src/napistu/network/net_propagation.py

```diff
@@ -0,0 +1,109 @@
+import pandas as pd
+import numpy as np
+import igraph as ig
+import inspect
+
+from typing import Optional
+
+def personalized_pagerank_by_attribute(
+    g: ig.Graph,
+    attribute: str,
+    damping: float = 0.85,
+    calculate_uniform_dist: bool = True,
+    additional_propagation_args: Optional[dict] = None,
+) -> pd.DataFrame:
+    """
+    Run personalized PageRank with reset probability proportional to a vertex attribute.
+    Optionally computes uniform PPR over nonzero attribute nodes.
+
+    Parameters
+    ----------
+    g : igraph.Graph
+        The input graph.
+    attribute : str
+        The vertex attribute to use for personalization.
+    damping : float, optional
+        Damping factor (default 0.85).
+    calculate_uniform_dist : bool, optional
+        If True, also compute uniform PPR over nonzero attribute nodes.
+    additional_propagation_args : dict, optional
+        Additional arguments to pass to igraph's personalized_pagerank. Keys must match the method's signature.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame with columns ['name', 'pagerank_by_attribute', attribute] and optionally 'pagerank_uniform'.
+
+    Example
+    -------
+    >>> import igraph as ig
+    >>> from scraps.utils import personalized_pagerank_by_attribute
+    >>> g = ig.Graph.Full(3)
+    >>> g.vs['name'] = ['A', 'B', 'C']
+    >>> g.vs['score'] = [1, 0, 2]
+    >>> df = personalized_pagerank_by_attribute(g, 'score')
+    >>> print(df)
+    """
+    # Validate and extract attribute (missing/None as 0)
+    attr = _ensure_nonnegative_vertex_attribute(g, attribute)
+
+    # Validate additional_propagation_args
+    if additional_propagation_args is None:
+        additional_propagation_args = {}
+    else:
+        valid_args = set(inspect.signature(g.personalized_pagerank).parameters.keys())
+        for k in additional_propagation_args:
+            if k not in valid_args:
+                raise ValueError(f"Invalid argument for personalized_pagerank: {k}")
+
+    # Personalized PageRank (no normalization, igraph handles it)
+    pr_attr = g.personalized_pagerank(reset=attr.tolist(), damping=damping, **additional_propagation_args)
+
+    # Node names
+    names = g.vs["name"] if "name" in g.vs.attributes() else list(range(g.vcount()))
```

Comment:
This seems like a convenient function that could be added as a helper to make sure it is consistently done in the library.
I also quickly checked and did not find a way to get this directly from igraph.